### DISCUSSION TITLE
Fulfillment Inbound API v2024-03-20 generatePlacementOptions (CustomPlacement details is only supported for inbound plans with destinations in IN), What's the problem?

### DISCUSSION THREAD

**Discussion by cwoinc** [amzn/selling-partner-api-models#3956] [General]
Parameters when requesting generatePlacementOptions:
```
{
  "customPlacement": [
    {
      "warehouseId": "ONT8",
      "items": [
        {
          "msku": "SKU1",
          "quantity": 50,
          "labelOwner": "SELLER",
          "prepOwner": "SELLER"
        }
      ]
    }
  ]
}
```

Error message returned when requesting generatePlacementOptions:
```
{
  "errors": [
    {
      "code": "BadRequest",
      "message": "ERROR: CustomPlacement details is only supported for inbound plans with destinations in IN.",
      "details": ""
    }
  ]
}
```

1. createInboundPlan
2. generatePackingOptions
3. listPackingOptions
4. listPackingGroupItems
5. confirmPackingOption
6. setPackingInformation
7. generatePlacementOptions

I used the US site to create a shipment plan and followed the above sequence step by step. When I reached generatePlacementOptions, the above error message appeared. If I give up on generatePlacementOptions and continue executing listPlacementOptions the result is empty (as shown in the JSON below). Why does this problem occur?

```
{"pagination":{},"placementOptions":[]}
```

**Comment by jackmin-sw** [comment]
@cwoinc hi, I am in the same flow, now I'm trying to call the generatePlacementOptions,
{{baseUrl}}/inbound/fba/2024-03-20/inboundPlans/:inboundPlanId/placementOptions
but the body customPlacement need a warehouseId, how can we get the warehouseId? when we create the inbound plan, we only know the marketplace id, could you help give some advice for this? I saw you put the warehouse id in your request

**Comment by cwoinc** [comment]
> @cwoinc hi, I am in the same flow, now I'm trying to call the generatePlacementOptions, {{baseUrl}}/inbound/fba/2024-03-20/inboundPlans/:inboundPlanId/placementOptions but the body customPlacement need a warehouseId, how can we get the warehouseId? when we create the inbound plan, we only know the marketplace id, could you help give some advice for this? I saw you put the warehouse id in your request

I'm very sorry, I don't know what the correct warehouse ID should be. This is an attempt to set a value. The target warehouse code of the Amazon shipment I set

**Comment by jackmin-sw** [comment]
> > @cwoinc hi, I am in the same flow, now I'm trying to call the generatePlacementOptions, {{baseUrl}}/inbound/fba/2024-03-20/inboundPlans/:inboundPlanId/placementOptions but the body customPlacement need a warehouseId, how can we get the warehouseId? when we create the inbound plan, we only know the marketplace id, could you help give some advice for this? I saw you put the warehouse id in your request
> 
> I'm very sorry, I don't know what the correct warehouse ID should be. This is an attempt to set a value. The target warehouse code of the Amazon shipment I set

thanks！

**Comment by cwoinc** [comment]
@shreeharsh-a What do you mean by closing this issue directly? If you think the way I asked the question is incorrect, you can point it out, but if you close the issue without saying anything, do you think that will solve the problem?

**Comment by shreeharsh-a** [comment]
Please reach out to developer support so that we can better help you with issues related to SP API.

`Note: SP API related issues / troubleshooting support is managed by a different team. We can help if the issue is related to the content published on this repo.`

**Comment by qdev89** [comment]
@cwoinc got same issue on call generatePlacementOptions with error:
`{
    "errors": [
        {
            "code": "BadRequest",
            "message": "ERROR: CustomPlacement details is only supported for inbound plans with destinations in IN.",
            "details": ""
        }
    ]
}`

I'm totally stuck and can't process further. Anybody please help.
Thanks

**Comment by cwoinc** [comment]
@qdev89 @jackmin-sw CustomPlacementInput is not required. No error will be reported if it is not filled in.(ps: The documentation for Fulfillment Inbound API v2024-03-20 has been updated)

**Comment by shreeharsh-a** [comment]
@MeenaAmz

**Comment by MeenaAmz** [comment]
CustomPlacementInput is an optional parameter while calling [generatePlacementOptions](https://developer-docs.amazon.com/sp-api/docs/fulfillment-inbound-api-v2024-03-20-reference#generateplacementoptions). Please check and provide us with request and response details if the issue is not resolved.
